### DISCUSSION TITLE
[Feature]: Make output$obj portable

### DISCUSSION THREAD

**Discussion by ChristineStawitz-NOAA** [NOAA-FIMS/FIMS#800] [Developers]
### Is your feature request related to a problem? Please describe.

In the NLL feature review, we discussed how @Andrea-Havron-NOAA tested and it isn't currently possible to export the rds file coming out of the model fit $obj such that you can re-run the model easily on another machine. @Cole-Monnahan-NOAA mentions this is a desirable feature.

### Describe the solution you would like.

export the rds file coming out of the model fit $obj such that you can re-run the model easily on another machine. 

### Describe alternatives you have considered

There's probably a way we could set up and call the model to make this happen, but it needs to be prioritized against other M2 work.

**Comment by kellijohnson-NOAA** [comment]
FIMS models also do not have access to the TMB function `retape()` which
```
  ## All external pointers are created in function "retape" and can be re-created
  ## by running retape() if e.g. the number of openmp threads is changed.
  ## set.defaults: reset internal parameters to their default values.
```
This lack of "retaping" limits the ability to pass around .RDS files (as noted above), to run MCMC chains in parallel, and simulate from models in parallel.

@msupernaw do you have any ideas on how to implement something for FIMS like what can be found [here for TMB](https://github.com/kaskr/adcomp/blob/628e97b9e8e23b2f4c5f80c165796a9649d3026f/TMB/R/TMB.R#L434)?

**Reply by kellijohnson-NOAA** [reply]
@Andrea-Havron-NOAA also brought up that retaping would be good or reference points.

**Comment by msupernaw** [comment]
Call clear before retape?

**Comment by Cole-Monnahan-NOAA** [comment]
Has someone tried this? I can put together a demo if that'd be helpful.

**Comment by Andrea-Havron-NOAA** [comment]
I've just run through a FIMS model in dev using the advanced user interface and was able to call `obj$retape()` on a model. I've also been able to successfully run a FIMS model in parallel using tmbstan, so it appears the retape issue has been resolved. We might want to retest the portability of obj to see if this issue has also been resolved.

**Reply by kellijohnson-NOAA** [reply]
Thank you @Andrea-Havron-NOAA for testing this 💪👍. Would we be able to create a test based on the code you ran? I think that we should continue to check for being able to retape in a reproducible way. Also, we should develop a test for the portability.

**Reply by Cole-Monnahan-NOAA** [reply]
That's amazing! An easy test would be to do like a jitter type thing where you parallelize N of re-optimizations from different starting values. Happy to help set up a test/demo.

**Reply by msupernaw** [reply]
@Andrea-Havron-NOAA This is a great revelation! This will save us a lot of time and tape space when it's time to compute reference points!
